go/consensus/tendermint/full/archive.go

```diff
@@ -0,0 +1,236 @@
+		return err
+	}
+
+	// Make sure the quit channel is closed when the node shuts down.
+	go func() {
+		select {
+		case <-srv.quitCh:
+		case <-srv.mux.Quit():
+			select {
+			case <-srv.quitCh:
+			default:
+				close(srv.quitCh)
+			}
+		}
+	}()
+
+	srv.Lock()
+	srv.isStarted = true
+	srv.Unlock()
+	close(srv.startedCh)
+
+	return nil
+}
+
+// Stop halts the service.
+func (srv *archiveService) Stop() {
+	if !srv.started() {
+		return
+	}
+	srv.stopOnce.Do(func() {
+		if err := srv.abciClient.Stop(); err != nil {
+			srv.Logger.Error("error on stopping abci client", "err", err)
+		}
+		srv.commonNode.Stop()
+	})
+}
+
+// Quit returns a channel that will be closed when the service terminates.
+func (srv *archiveService) Quit() <-chan struct{} {
+	return srv.quitCh
+}
+
+// Implements Backend.
+func (srv *archiveService) Synced() <-chan struct{} {
+	// Archive node is always considered synced.
+	ch := make(chan struct{})
+	close(ch)
+	return ch
+}
+
+// Implements Backend.
+func (srv *archiveService) EstimateGas(ctx context.Context, req *consensusAPI.EstimateGasRequest) (transaction.Gas, error) {
+	return 0, consensusAPI.ErrUnsupported
+}
+
+// Implements Backend.
+func (srv *archiveService) GetSignerNonce(ctx context.Context, req *consensusAPI.GetSignerNonceRequest) (uint64, error) {
+	return 0, consensusAPI.ErrUnsupported
+}
+
+// New creates a new archive-only consensus service.
+func NewArchive(
+	ctx context.Context,
+	dataDir string,
+	identity *identity.Identity,
+	genesisProvider genesisAPI.Provider,
+) (consensusAPI.Backend, error) {
+	var err error
+
+	srv := &archiveService{
+		commonNode: commonNode{
+			BaseBackgroundService: *cmservice.NewBaseBackgroundService("tendermint"),
+			ctx:                   ctx,
+			rpcCtx:                &tmrpctypes.Context{},
+			identity:              identity,
+			dataDir:               dataDir,
+			svcMgr:                cmbackground.NewServiceManager(logging.GetLogger("tendermint/servicemanager")),
+			startedCh:             make(chan struct{}),
+		},
+		startedCh: make(chan struct{}),
+		quitCh:    make(chan struct{}),
+	}
+	// Common node needs access to parent struct for initializing consensus services.
+	srv.commonNode.parentNode = srv
+
+	doc, err := genesisProvider.GetGenesisDocument()
+	if err != nil {
+		return nil, fmt.Errorf("tendermint/archive: failed to get genesis document: %w", err)
+	}
+	srv.genesis = doc
+
+	appConfig := &abci.ApplicationConfig{
+		DataDir:        filepath.Join(srv.dataDir, tmcommon.StateDir),
+		StorageBackend: db.GetBackendName(),
+		Pruning: abci.PruneConfig{
+			Strategy:      abci.PruneNone,
+			PruneInterval: time.Hour * 100, // Irrelevant as pruning is disabled.
+		},
+		HaltEpochHeight:     srv.genesis.HaltEpoch,
+		OwnTxSigner:         srv.identity.NodeSigner.Public(),
+		DisableCheckpointer: true,
+		InitialHeight:       uint64(srv.genesis.Height),
+		// ReadOnly should actually be preferable for archive node but this fails with:
```

Comment:
This is a badger issue (https://discuss.dgraph.io/t/read-only-log-truncate-required-to-run-db/16444/2).